lighthouse-core/gather/driver.js

```diff
@@ -115,15 +116,65 @@ class Driver {
   }
 
   /**
+   * Debounce enabling or disabling domains to prevent driver users from
+   * stomping on each other. Maintains an internal count of the times a domain
+   * has been enabled. Returns false if the command would have no effect (domain
+   * is already enabled or disabled), or if command would interfere with another
+   * user of that domain (e.g. two gatherers have enabled a domain, both need to
+   * disable it for it to be disabled). Returns true otherwise.
+   * @param {string} domain
+   * @param {string} action
+   * @return {boolean}
+   * @private
+   */
+  _checkDomainToggle(domain, action) {
+    const enable = action === 'enable';
+    const enabledCount = this._domainEnabledCounts.get(domain) || 0;
+    const newCount = enabledCount + (enable ? 1 : -1);
+    this._domainEnabledCounts.set(domain, Math.max(0, newCount));
+
+    // Switching to enabled or disabled, respectively.
+    if ((enable && newCount === 1) || (!enable && newCount === 0)) {
+      log.verbose('Driver', `${domain}.${action}`);
+      return true;
+    } else {
+      if (newCount < 0) {
+        log.error('Driver', `Attempted to disable domain '${domain}' when already disabled.`);
+      }
+      return false;
+    }
+  }
+
+  /**
    * Call protocol methods
    * @param {!string} method
    * @param {!Object} params
    * @return {!Promise}
    */
   sendCommand(method, params) {
+    const domainCommand = /^(\w+)\.(enable|disable)$/.exec(method);
+    if (domainCommand && !this._checkDomainToggle(domainCommand[1], domainCommand[2])) {
+      return Promise.resolve();
+    }
+
     return this._connection.sendCommand(method, params);
   }
 
+  /**
+   * Returns whether a domain is currently enabled.
+   * @param {string} domain
+   * @return {boolean}
+   */
+  domainEnabled(domain) {
```

Comment:
just curious here, what are your thoughts in general on `isX` prefixing for all boolean values/functions?